snagsby/sources.py

```diff
@@ -131,21 +137,30 @@ def get_raw_data(self):
         return json.loads(obj.decode())
 
 
+registry = Registry()
+registry.register_handler('s3', S3Source)
+registry.register_handler('sm', SMSource)
+
+
+def get_source(source):
+    source_type = urlparse(source).scheme
+    try:
+        return registry.get_handler(source_type)(source)
+    except KeyError:
+        logger.debug('Sources must be one of: {}'.format(
+            ", ".join(registry.get_names())))
```

Comment:
I'd prefer to have this explicitly return None, since the caller is relying on the behavior, but I'm not sure if our style guide requires it?